test/gtest/ucp/ucp_test.h

```diff
@@ -74,7 +79,7 @@ class ucp_test_base : public ucs::test_base {
     protected:
         ucs::handle<ucp_context_h> m_ucph;
         std::vector<ucs::handle<ucp_worker_h> >  m_workers;
-        std::vector<ucs::handle<ucp_ep_h> >      m_eps;
+        ep_map_t m_eps;
```

Comment:
What's the main reason - performance? It will add complexity (from readability point of view) for addressing workers/eps
to access `[i]` worker:
`m_workers[i]->first` vs current: `m_workers[i]`
to access `[i][j]` ep:
`m_workers[i]->second[j]` vs current: `m_eps[m_workers[i]][j]`

Also, if you don't need to connect all to all indexes you may have not initialized memory, map always contains only initialized instances.